File: problem-solving/Easy/5_plus-minus/python/main.py
# URL : https://www.hackerrank.com/challenges/plus-minus/problem

def plusMinus(arr):
    positive = negative = zero = 0
    if 0 < len(arr) <= 100:
        for i in arr:
            if -100 <= i <= 100:
                if i < 0:
                    negative += 1
                elif i == 0:
                    zero += 1
                else:
                    positive += 1

    print(positive/len(arr))
    print(negative/len(arr))
    print(zero/len(arr))


# test
plusMinus([-4, 3, -9, 0, 4, 1])

File: problem-solving/Easy/5_plus-minus/javascript/main.js
// URL : https://www.hackerrank.com/challenges/plus-minus/problem

// This is works on here but not working on hackerrank.

function plusMinus(input) {
  var lines = input.toString().split("\n"),
    size = parseInt(lines.shift()),
    pos = 0,
    neg = 0,
    zero = 0;
  lines[0].split(" ").map(function (p) {
    val = parseInt(p);
    if (val < 0) neg++;
    else if (val == 0) zero++;
    else pos++;
  });
  console.log((pos / size).toFixed(6));
  console.log((neg / size).toFixed(6));
  console.log((zero / size).toFixed(6));
}
//test
inp = `10
0 100 35 0 94 40 42 87 59 0`;
plusMinus(inp);
